### PR TITLE
Document css and tidy up lint

### LINKED ISSUE #1056: Document some simple guidelines for our approach to CSS
A few times the question of whether we should adopt some kind of methodology for CSS has come up. If we go this route here's some likely goals of adopting anything along these lines:
- Simplicity - it can't be too complex or we just won't use it.
- Consistency - consistency is king. 
- Sane specificity. E.g. avoiding the kind of crazy specifity hacks we needed to do in the style-refresh.

There are lots of methodologies that exist e.g. ([SMACSS](https://smacss.com/), [BEM](http://getbem.com/) and [suitcss](https://github.com/suitcss/suit/blob/master/doc/naming-conventions.md)). Most look to provide some structure to classnames to aid consistent code modularization.

We don't have to opt for any one specific approach necessarily, but whatever we decide would be good to document how we are going to build our component classnames (or if we are wholesale adopting something that exists we can just point to their docs instead).

Bonus points for stylelint rules to help maintain whatever the end result is.

### PR DESCRIPTION
**Pull request by tofumatt**
Fixes #1056.

### CODE REVIEW

**Comment by tofumatt** [comment]
r?

Docs are here: https://github.com/mozilla/addons-frontend/pull/2143/commits/86ac2fe2541df5dbcb8038deb7af609aa06192bd

Rest is just aligning with some saner SuitCSS rules and alphabetising, really.

File: src/amo/components/SearchSort/SearchSort.scss [hunk 1]
```diff
@@ -1,18 +1,18 @@
...
+  border-radius: $border-radius-default;
   margin: $padding-page ($padding-page * 2);
   overflow: hidden;
 }
 
 .SearchSort-toggle {
   @include button($background: $base-color, $color: $link-color);
 
-  // We do this so the button doesn't have weird edges on hover when the
+  // We do this so the button doesn"t have weird edges on hover when the
```
Comment (kumar303) [inline src/amo/components/SearchSort/SearchSort.scss]: find/replace casualty

File: src/amo/css/AddonReview.scss [hunk 2]
```diff
@@ -17,33 +17,33 @@
...
-  text-align: left;
-  width: 100%;
-  padding: 0.7em;
+.AddonReview-submit,
+.AddonReview-textarea {
   border: 2px solid $form-border-color;
   border-radius: 8px;
   box-shadow: none;
+  margin-bottom: 12px;
+  padding: 0.7em;
```
Comment (kumar303) [inline src/amo/css/AddonReview.scss]: I guess `em` is allowed but just discouraged?

File: src/amo/css/ScreenShots.scss [hunk 3]
```diff
@@ -38,7 +38,7 @@ $screenshot-width: 320px;
   width: $screenshot-width;
 }
 
-// Don't stretch the images.
+// Don"t stretch the images.
```
Comment (kumar303) [inline src/amo/css/ScreenShots.scss]: another one

File: src/core/css/inc/vars.scss [hunk 4]
```diff
@@ -45,6 +45,6 @@ $border-radius-default: 9px;
 $border-radius-s: 4px;
 $border-radius-m: 8px;
 
-// Page padding (this is the padding for an entire page that isn't full-width,
+// Page padding (this is the padding for an entire page that isn"t full-width,
```
Comment (kumar303) [inline src/core/css/inc/vars.scss]: 🍟

File: src/core/css/ErrorHandler.scss [hunk 5]
```diff
@@ -0,0 +1,15 @@
+@import "~core/css/inc/vars";
```
Comment (kumar303) [inline src/core/css/ErrorHandler.scss]: I'm confused about this file. Did you mean to delete some code from [ui/components/ErrorList/styles.scss](https://github.com/mozilla/addons-frontend/blob/86ac2fe2541df5dbcb8038deb7af609aa06192bd/src/ui/components/ErrorList/styles.scss) too?

File: docs/css.md [hunk 6]
```diff
@@ -0,0 +1,11 @@
...
+* **Do not import directly from `~core/`**: each app has its own `mixins.scss`, `vars.scss`, etc. Use `~amo/css/mixins.scss` instead of `~core/css/mixins.scss`. Each app extends core but be careful of changing things in core; it's best to put overrides in an app's mixin.
+* **Use `px` (pixels) for size instead of `em`, `rem`, etc.**: pixel sizes work fine and keeps things consistent. If you want to scale something with screen size `%` (percent) values are okay.
+* **Keep CSS local to the component:** don't use `App.scss` or `SearchPage/style.scss` to style a button component on a `SearchPage` component. Instead create options that can be passed to differently style the button from the `SearchPage` component.
+* **Don't nest selectors; use more specific class names instead**: prefer `<ul  className="List"><li className="List-item"></li></ul>` and the selector `.List-item` over `.List li`. This seems verbose but it prevents messing nesting bugs and confusion as the codebase grows and ensures easier organization and scanning of CSS rules.
```
Comment (kumar303) [inline docs/css.md]: typo: ...messing nesting bugs...

**Comment by muffinresearch** [comment]
@tofumatt not a priority at all, but it would be nice to get this fixed up and landed so it doesn't bitrot further.

**Comment by muffinresearch** [comment]
@tofumatt Would be nice to clean this up and land it.

**Comment by tofumatt** [comment]
Alright! Redid this and fixed the issues with search/replace. I think the issues are all fixed, r?

**Comment by kumar303** [comment]
Looks like this has a conflict

**Comment by tofumatt** [comment]
Sorry about that, the greenkeeper and style PRs have been messing with this. Rebased and ready (for now 😅 )!

File: docs/css.md [hunk 7]
```diff
@@ -0,0 +1,11 @@
...
+* **Keep CSS local to the component:** don't use `App.scss` or `SearchPage/style.scss` to style a button component on a `SearchPage` component. Instead create options that can be passed to differently style the button from the `SearchPage` component.
+* **Don't nest selectors; use more specific class names instead**: prefer `<ul  className="List"><li className="List-item"></li></ul>` and the selector `.List-item` over `.List li`. This seems verbose but it prevents nesting bugs and confusion as the codebase grows. It also makes it easier to scan CSS rules at a glance.
+* **When overrides make sense, keep them in the parent component and use CSS nesting:** this is the only time we use nesting instead of specifically-named class selectors. There are a few cases where tiny adjustments may be made to `margin`, etc. for a child component in only one context. If modifying the `margin` of `Result` inside `SearchResultList`, it is appropriate to `.SearchResultList .Result` the `SearchResultList/style.scss`.
```
Comment (kumar303) [inline docs/css.md]: typo: ...it is appropriate to declare `.SearchResultList .Result` in `SearchResultList/style.scss`

File: src/core/css/inc/mixins.scss [hunk 8]
```diff
@@ -70,23 +70,27 @@
   cursor: not-allowed;
 }
 
-/* stylelint-disable at-rule-empty-line-before, media-feature-no-missing-punctuation  */
+/*
+  stylelint-disable at-rule-empty-line-before,
+  media-feature-no-missing-punctuation
```
Comment (kumar303) [inline src/core/css/inc/mixins.scss]: does this still work when broken into a new line? Well, if the lint rules aren't failing I guess it does. Is the override still needed?

File: src/ui/components/Button/Button.scss [hunk 9]
```diff
@@ -1,53 +1,28 @@
...
+@import "~core/css/inc/mixins";
+@import "~ui/css/vars";
 
 .Button--neutral {
-  @include button(
-    $background: $neutral-base-color,
-    $background-active: $neutral-active-color,
-    $background-hover: $neutral-hover-color
-  );
+  @include button($background: $neutral-base-color, $background-active: $neutral-active-color, $background-hover: $neutral-hover-color);
```
Comment (kumar303) [inline src/ui/components/Button/Button.scss]: Do all of these need to go on one line? I found it easier to read when they were split across four lines. It's also more friendly to split screen editors.

Comment (tofumatt) [inline src/ui/components/Button/Button.scss]: You're totally right, not sure why I did this. I'll change it back 👍

**Pull request merged**